[PATCH] do not hardcoded libdir

Upstream-Status: Pending

libdir is hardcoded to ${exec_prefix}/lib, but we want it to support multilib

Signed-off-by: Roy.Li <rongqing.li@windriver.com>
---
 src/dynamic-examples/dynamic-preprocessor/Makefile.am | 2 +-
 src/dynamic-examples/dynamic-rule/Makefile.am         | 2 +-
 src/dynamic-plugins/sf_engine/Makefile.am             | 2 +-
 src/dynamic-preprocessors/dcerpc2/Makefile.am         | 2 +-
 src/dynamic-preprocessors/dnp3/Makefile.am            | 2 +-
 src/dynamic-preprocessors/dns/Makefile.am             | 2 +-
 src/dynamic-preprocessors/file/Makefile.am            | 2 +-
 src/dynamic-preprocessors/ftptelnet/Makefile.am       | 2 +-
 src/dynamic-preprocessors/gtp/Makefile.am             | 2 +-
 src/dynamic-preprocessors/imap/Makefile.am            | 2 +-
 src/dynamic-preprocessors/modbus/Makefile.am          | 2 +-
 src/dynamic-preprocessors/pop/Makefile.am             | 2 +-
 src/dynamic-preprocessors/reputation/Makefile.am      | 2 +-
 src/dynamic-preprocessors/rzb_saac/Makefile.am        | 2 +-
 src/dynamic-preprocessors/sdf/Makefile.am             | 2 +-
 src/dynamic-preprocessors/sip/Makefile.am             | 2 +-
 src/dynamic-preprocessors/smtp/Makefile.am            | 2 +-
 src/dynamic-preprocessors/ssh/Makefile.am             | 2 +-
 src/dynamic-preprocessors/ssl/Makefile.am             | 2 +-
 19 files changed, 19 insertions(+), 19 deletions(-)

diff --git a/src/dynamic-examples/dynamic-preprocessor/Makefile.am b/src/dynamic-examples/dynamic-preprocessor/Makefile.am
index 255755e..1cafead 100644
--- a/src/dynamic-examples/dynamic-preprocessor/Makefile.am
+++ b/src/dynamic-examples/dynamic-preprocessor/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include
 
-noinst_libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+noinst_libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 noinst_lib_LTLIBRARIES = lib_sfdynamic_preprocessor_example.la
 
diff --git a/src/dynamic-examples/dynamic-rule/Makefile.am b/src/dynamic-examples/dynamic-rule/Makefile.am
index e5338d2..82eb212 100644
--- a/src/dynamic-examples/dynamic-rule/Makefile.am
+++ b/src/dynamic-examples/dynamic-rule/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include
 
-noinst_libdir = ${exec_prefix}/lib/snort_dynamicrules
+noinst_libdir = ${exec_prefix}${base_libdir}/snort_dynamicrules
 
 noinst_lib_LTLIBRARIES = lib_sfdynamic_example_rule.la
 
diff --git a/src/dynamic-plugins/sf_engine/Makefile.am b/src/dynamic-plugins/sf_engine/Makefile.am
index d6a6200..3d270c3 100644
--- a/src/dynamic-plugins/sf_engine/Makefile.am
+++ b/src/dynamic-plugins/sf_engine/Makefile.am
@@ -5,7 +5,7 @@ INCLUDES = @INCLUDES@
 
 lib_LTLIBRARIES = libsf_engine.la
 
-libdir = ${exec_prefix}/lib/snort_dynamicengine
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicengine
 
 libsf_engine_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
 
diff --git a/src/dynamic-preprocessors/dcerpc2/Makefile.am b/src/dynamic-preprocessors/dcerpc2/Makefile.am
index 9843073..8a0caa5 100644
--- a/src/dynamic-preprocessors/dcerpc2/Makefile.am
+++ b/src/dynamic-preprocessors/dcerpc2/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs -I$(srcdir)/includes
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_dce2_preproc.la
 
diff --git a/src/dynamic-preprocessors/dnp3/Makefile.am b/src/dynamic-preprocessors/dnp3/Makefile.am
index 1358862..18e2178 100644
--- a/src/dynamic-preprocessors/dnp3/Makefile.am
+++ b/src/dynamic-preprocessors/dnp3/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_dnp3_preproc.la
 
diff --git a/src/dynamic-preprocessors/dns/Makefile.am b/src/dynamic-preprocessors/dns/Makefile.am
index f22d90e..49d2f55 100644
--- a/src/dynamic-preprocessors/dns/Makefile.am
+++ b/src/dynamic-preprocessors/dns/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_dns_preproc.la
 
diff --git a/src/dynamic-preprocessors/file/Makefile.am b/src/dynamic-preprocessors/file/Makefile.am
index d71c0fd..834f7ac 100644
--- a/src/dynamic-preprocessors/file/Makefile.am
+++ b/src/dynamic-preprocessors/file/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs -I./include   
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_file_preproc.la
 
diff --git a/src/dynamic-preprocessors/ftptelnet/Makefile.am b/src/dynamic-preprocessors/ftptelnet/Makefile.am
index fe73835..3457231 100644
--- a/src/dynamic-preprocessors/ftptelnet/Makefile.am
+++ b/src/dynamic-preprocessors/ftptelnet/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs -I$(srcdir)/includes
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_ftptelnet_preproc.la
 
diff --git a/src/dynamic-preprocessors/gtp/Makefile.am b/src/dynamic-preprocessors/gtp/Makefile.am
index 51c961a..d7fb9b2 100644
--- a/src/dynamic-preprocessors/gtp/Makefile.am
+++ b/src/dynamic-preprocessors/gtp/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs -I$(srcdir)/includes
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_gtp_preproc.la
 
diff --git a/src/dynamic-preprocessors/imap/Makefile.am b/src/dynamic-preprocessors/imap/Makefile.am
index 533e43d..9efcbc4 100644
--- a/src/dynamic-preprocessors/imap/Makefile.am
+++ b/src/dynamic-preprocessors/imap/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_imap_preproc.la
 
diff --git a/src/dynamic-preprocessors/modbus/Makefile.am b/src/dynamic-preprocessors/modbus/Makefile.am
index 05fc63e..e4f1415 100644
--- a/src/dynamic-preprocessors/modbus/Makefile.am
+++ b/src/dynamic-preprocessors/modbus/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_modbus_preproc.la
 
diff --git a/src/dynamic-preprocessors/pop/Makefile.am b/src/dynamic-preprocessors/pop/Makefile.am
index d4b07b3..1345bc6 100644
--- a/src/dynamic-preprocessors/pop/Makefile.am
+++ b/src/dynamic-preprocessors/pop/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_pop_preproc.la
 
diff --git a/src/dynamic-preprocessors/reputation/Makefile.am b/src/dynamic-preprocessors/reputation/Makefile.am
index 4e16142..14ba0cb 100644
--- a/src/dynamic-preprocessors/reputation/Makefile.am
+++ b/src/dynamic-preprocessors/reputation/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs -I$(srcdir)/includes
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_reputation_preproc.la
 
diff --git a/src/dynamic-preprocessors/rzb_saac/Makefile.am b/src/dynamic-preprocessors/rzb_saac/Makefile.am
index f0b0912..6ebe04e 100644
--- a/src/dynamic-preprocessors/rzb_saac/Makefile.am
+++ b/src/dynamic-preprocessors/rzb_saac/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-pkglibdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+pkglibdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 pkglib_LTLIBRARIES = sf_rzb_saac_preproc.la
 
diff --git a/src/dynamic-preprocessors/sdf/Makefile.am b/src/dynamic-preprocessors/sdf/Makefile.am
index 97b416e..04b5ce5 100644
--- a/src/dynamic-preprocessors/sdf/Makefile.am
+++ b/src/dynamic-preprocessors/sdf/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_sdf_preproc.la
 
diff --git a/src/dynamic-preprocessors/sip/Makefile.am b/src/dynamic-preprocessors/sip/Makefile.am
index 67ac1e1..b4f360f 100644
--- a/src/dynamic-preprocessors/sip/Makefile.am
+++ b/src/dynamic-preprocessors/sip/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs -I$(srcdir)/includes
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_sip_preproc.la
 
diff --git a/src/dynamic-preprocessors/smtp/Makefile.am b/src/dynamic-preprocessors/smtp/Makefile.am
index 3b5893f..96750df 100644
--- a/src/dynamic-preprocessors/smtp/Makefile.am
+++ b/src/dynamic-preprocessors/smtp/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_smtp_preproc.la
 
diff --git a/src/dynamic-preprocessors/ssh/Makefile.am b/src/dynamic-preprocessors/ssh/Makefile.am
index 0eab645..5ae9b80 100644
--- a/src/dynamic-preprocessors/ssh/Makefile.am
+++ b/src/dynamic-preprocessors/ssh/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_ssh_preproc.la
 
diff --git a/src/dynamic-preprocessors/ssl/Makefile.am b/src/dynamic-preprocessors/ssl/Makefile.am
index 4ac2943..8d3abe3 100644
--- a/src/dynamic-preprocessors/ssl/Makefile.am
+++ b/src/dynamic-preprocessors/ssl/Makefile.am
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${exec_prefix}${base_libdir}/snort_dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_ssl_preproc.la
 
-- 
1.9.1

